commands/volumes/volume-stop.go

```diff
@@ -4,13 +4,61 @@ import (
 	"net/http"
 
 	"github.com/gluster/glusterd2/errors"
+	"github.com/gluster/glusterd2/gdctx"
 	"github.com/gluster/glusterd2/rest"
+	"github.com/gluster/glusterd2/transaction"
 	"github.com/gluster/glusterd2/volume"
 
 	log "github.com/Sirupsen/logrus"
 	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
 )
 
+func stopBricks(c transaction.TxnCtx) error {
+	var volname string
+	if e := c.Get("volname", &volname); e != nil {
+		c.Logger().WithFields(log.Fields{
+			"error": e,
+			"key":   "volname",
+		}).Error("failed to get value for key from context")
+		return e
+	}
+
+	vol, e := volume.GetVolume(volname)
+	if e != nil {
+		// this shouldn't happen
+		c.Logger().WithFields(log.Fields{
+			"error":   e,
+			"volname": volname,
+		}).Error("failed to get volinfo for volume")
+		return e
+	}
+
+	for _, brick := range vol.Bricks {
+		if uuid.Equal(brick.ID, gdctx.MyUUID) {
+			c.Logger().WithFields(log.Fields{
+				"volume": volname,
+				"brick":  brick.Hostname + ":" + brick.Path,
+			}).Info("would stop brick")
+			//TODO: Stop running brick processes once the daemon management package is ready
+		}
+	}
+	return nil
+}
+
+func registerVolStopStepFuncs() {
+	transaction.RegisterStepFunc(stopBricks, "vol-stop.Commit")
```

Comment:
This has no rollback. I think it make sense. But If one of the nodes in unreachable ?